### PR TITLE
Fix invalid links

### PR DESCRIPTION
**Pull request by byyue**
To pass the HTML Proofer

### CODE REVIEW

File: _scala3-reference/other-new-features/explicit-nulls.md [hunk 1]
```diff
@@ -537,4 +537,4 @@ Our strategy for binary compatibility with Scala binaries that predate explicit
 and new libraries compiled without `-Yexplicit-nulls` is to leave the types unchanged
 and be compatible but unsound.
 
-[More details](https://dotty.epfl.ch/docs/internals/explicit-nulls.html)
+[More details](https://docs.scala-lang.org/scala3/reference/other-new-features/explicit-nulls.html)
```
Comment (bishabosha) [inline _scala3-reference/other-new-features/explicit-nulls.md:540]: this link is to the current page - perhaps we can link to the GitHub doc itself as it doesn't appear to be published anywhere: https://github.com/lampepfl/dotty/blob/main/docs/_docs/internals/explicit-nulls.md

File: _overviews/scaladoc/contribute.md [hunk 2]
```diff
@@ -25,4 +25,4 @@ which covers the steps and workflow necessary work on the Scaladoc tool.
 As of Scala 2.13, the Scaladoc tool is maintained but not actively
 developed.  Major development of Scaladoc will progress as a part of
 Dotty for Scala 3 in the
-[Dottydoc](https://dotty.epfl.ch/docs/usage/dottydoc.html) tool.
+[Dottydoc](https://dotty.epfl.ch/3.0.0/docs/usage/dottydoc.html) tool.
```
Comment (julienrf) [inline _overviews/scaladoc/contribute.md:28]: I suggest we replace this link with an internal link to:

```suggestion
[Scaladoc]({% link _overviews/scala3-scaladoc/index.md %}) tool.
```

File: scala3/scaladoc.md [hunk 3]
```diff
@@ -81,5 +81,5 @@ If you encounter a bug or have an idea for improvement, don't hesitate to create
 
 [scaladoc-docstrings]: https://dotty.epfl.ch/docs/usage/scaladoc/docstrings.html
 [static-documentation]: https://dotty.epfl.ch/docs/usage/scaladoc/static-site.html
-[built-in-blog]: https://dotty.epfl.ch/docs/usage/scaladoc/blog.html
+[built-in-blog]: https://dotty.epfl.ch/3.0.0/docs/usage/scaladoc/blog.html
```
Comment (julienrf) [inline scala3/scaladoc.md:84]:
```suggestion
[built-in-blog]: {% link _overviews/scala3-scaladoc/blog.md %}
```

File: scala3/scaladoc.md [hunk 4]
```diff
@@ -81,5 +81,5 @@ If you encounter a bug or have an idea for improvement, don't hesitate to create
 
 [scaladoc-docstrings]: https://dotty.epfl.ch/docs/usage/scaladoc/docstrings.html
 [static-documentation]: https://dotty.epfl.ch/docs/usage/scaladoc/static-site.html
```
Comment (julienrf) [inline scala3/scaladoc.md:83]:
```suggestion
[scaladoc-docstrings]: {% link _overviews/scala3-scaladoc/docstrings.md %}
[static-documentation]: {% link _overviews/scala3-scaladoc/static-site.md %}
```

**Review by julienrf — COMMENTED** [review]
Thank you for helping! I suggest we link to internal pages (in the docs.scala-lang website) as most as possible.

**Review by julienrf — APPROVED** [review]

**Pull request merged**